rotkehlchen/chain/evm/decoding/pendle/decoder.py

```diff
@@ -1,122 +1,317 @@
+                    self.evm_inquirer.native_token
+                    if (addr := bytes_to_address(tx_log.data[32:64])) == ETH_SPECIAL_ADDRESS
+                    else self.base.get_or_create_evm_token(addr)
+                )
+                amount_in = asset_normalized_value(
+                    amount=int.from_bytes(tx_log.data[128:160]),
+                    asset=token_in,
+                )
+                break
+        else:
+            log.error(f'Could not retrieve swap details for {swap_type=} in transaction {context.transaction}')  # noqa: E501
+            return DEFAULT_DECODING_OUTPUT
+
+        in_event, out_event = None, None
+        for event in context.decoded_events:
+            if (
+                    event.asset == token_out and
+                    event.amount == amount_out and
+                    event.address == PENDLE_ROUTER_ADDRESS and
+                    event.event_type in (HistoryEventType.SPEND, HistoryEventType.TRADE) and
+                    event.event_subtype in (HistoryEventSubType.NONE, HistoryEventSubType.SPEND)
+            ):
+                out_event = event
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.notes = f'Swap {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} in Pendle'  # noqa: E501
+
+            elif (
+                    event.asset == token_in and
+                    event.amount == amount_in and
+                    event.address == PENDLE_ROUTER_ADDRESS and
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+            ):
+                in_event = event
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.notes = f'Receive {event.amount} {token_in.symbol} from Pendle swap'
+
+        # prepare action item for scenarios where receive event is missing
+        action_items = []
+        if in_event is None and out_event is not None:
+            action_items.append(ActionItem(
+                action='transform',
+                from_event_type=HistoryEventType.RECEIVE,
+                from_event_subtype=HistoryEventSubType.NONE,
+                to_event_type=HistoryEventType.TRADE,
+                to_event_subtype=HistoryEventSubType.RECEIVE,
+                amount=amount_in,
+                asset=token_in,
+                paired_events_data=([out_event], True),
+                to_notes=f'Receive {amount_in} {token_in.symbol} from Pendle swap',
+                to_counterparty=CPT_PENDLE,
+            ))
+
+        maybe_reshuffle_events(
+            ordered_events=[out_event, in_event],
+            events_list=context.decoded_events,
+        )
```

Comment:
you need to reshuffle only if you found both of them right?

Reply:
this wasn't answered. I believe you should have the reshuffle in a `elif` clause since if you only got one of the events you are effectively reshuffling twice